numpy/linalg/_linalg.pyi

```diff
@@ -480,3 +487,27 @@ def matmul(
     x1: _ArrayLikeComplex_co,
     x2: _ArrayLikeComplex_co,
 ) -> NDArray[complexfloating]: ...
+
+@overload
+def geneigvals(a: _ArrayLikeInt_co, b: _ArrayLikeInt_co) -> NDArray[float64]: ...
+@overload
+def geneigvals(a: _ArrayLikeFloat_co, b: _ArrayLikeFloat_co) -> NDArray[floating]: ...
+@overload
+def geneigvals(a: _ArrayLikeComplex_co, b: _ArrayLikeComplex_co) -> NDArray[complexfloating]: ...
```

Comment:
at runtime the parameters are called `A` and `B`, but here they are `a` and `b`. That would cause code like `geneigvals(A=arr1, B=arr2)` to be rejected by static type-checkers, even though it would work at runtime. Conversely, `geneigvals(a=arr1, b=arr2)` would be accepted by static type-checkers, but will raise an error at runtime.

So consider making them positional only, or otherwise ensure that their names match.